src/graphics/opengl_wrapper.cpp

```diff
@@ -789,9 +790,120 @@ void render_new_text(sys::state& state, text::stored_glyphs const& txt, color_mo
 	internal_text_render(state, txt, x, y + size, size, f, subroutines, icon_subroutines);
 }
 
+void render_classic_text(sys::state& state, float x, float y, text::stored_glyphs const& txt, color_modification enabled, color3f const& c, text::bm_font const& font) {
+	float adv = 1.0f / font.width; // Font texture atlas spacing.
 
-void render_text(sys::state& state, text::stored_glyphs const& txt, color_modification enabled, float x, float y,
-		color3f const& c, uint16_t font_id) {
+	bind_vertices_by_rotation(state, ui::rotation::upright, false);
+
+	GLuint subroutines[2] = { map_color_modification_to_index(enabled), parameters::subsprite_b };
+	glUniformSubroutinesuiv(GL_FRAGMENT_SHADER, 2, subroutines);
+
+	// Set Text Color, all one color for now.
+
+	//------ FOR SCHOMBERT ------//
+	// Every iteration of this loop draws one character of the string 'fmt'.
+	//'texlst' contains information for each vertex of each rectangle for each character.
+	// Every 4 elements in 'texlst' is one complete rectangle, and one character.
+	//'texlst[i].texx' and 'texlst[i].texy' are the intended texture coordinates of a vertex on the texture.
+	//'texlst[i].x' and 'texlst[i].y' are the coordinates of a vertex of the rendered rectangle in the window.
+	// The color variables are unused currently.
+	//
+	// Spacing, kearning, etc. are already applied.
+	// Scaling (unintentionally) is also applied (by whatever part of Alice scales the normal fonts).
+
+	glActiveTexture(GL_TEXTURE0);
+	glBindTexture(GL_TEXTURE_2D, font.ftexid);
+
+	for(uint32_t i = 0; i < txt.glyph_count; ++i) {
+		if(uint8_t(txt.glyph_info[i].codepoint) == '@') {
```

Comment:
codepoint isn't the utf32 value, it is the internal font glyph id, which has not intrinsic relation to the codepoint. Also, you would still need to convert it back to win1252 at some point

Reply:
fixed